security_monkey/auditors/sqs.py

```diff
@@ -0,0 +1,107 @@
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+"""
+.. module: security_monkey.auditors.sqs
+    :platform: Unix
+
+.. version:: $$VERSION$$
+.. moduleauthor:: Denver Janke <denverjanke@gmail.com>
+
+"""
+
+from security_monkey.auditor import Auditor
+from security_monkey.watchers.sqs import SQS
+from security_monkey.exceptions import InvalidARN
+from security_monkey.exceptions import InvalidSourceOwner
+from security_monkey.datastore import Account
+
+import re
+
+
+class SQSAuditor(Auditor):
+    index = SQS.index
+    i_am_singular = SQS.i_am_singular
+    i_am_plural = SQS.i_am_plural
+
+    def __init__(self, accounts=None, debug=False):
+        super(SQSAuditor, self).__init__(accounts=accounts, debug=debug)
+
+    def check_sqstopicpolicy_crossaccount(self, sqsitem):
+        """
+        alert on cross account access
+        """
+        policy = sqsitem.config
+        for statement in policy.get("Statement", []):
+            account_numbers = []
+            account_number = ''
+            princ = statement.get("Principal", {})
+            if isinstance(princ, dict):
+                princ_aws = princ.get("AWS", "error")
+            else:
+                princ_aws = princ
+            if princ_aws == "*":
+                topic_arn = statement.get("Condition", {}) \
+                    .get("ArnEquals", {}) \
+                    .get("AWS:SourceArn", None)
+                if not topic_arn:
+                    tag = "SQS Topic open to everyone"
+                    notes = "An SQS policy where { 'Principal': { 'AWS': '*' } } must also have"
+                    notes += " a {'Condition': {'ArnEquals': { 'AWS:SourceArn': '<ACCOUNT_NUMBER>' } } }"
+                    notes += " or it is open to the world. In this case, anyone is allowed to perform "
+                    notes += " this action(s): {}".format(statement.get("Action"))
+                    self.add_issue(10, tag, sqsitem, notes=notes)
+                    continue
+                else:
+
+                    try:
+                        account_numbers.append(str(re.search('arn:aws:sns:[a-z-]+-\d:([0-9-]+):', topic_arn).group(1)))
+                    except:
+                        raise InvalidARN(topic_arn)
+
+            else:
+                if isinstance(princ_aws, list):
+                    for entry in princ_aws:
+                        account_numbers.append(str(entry))
+                else:
+                    try:
+                        account_numbers.append(str(princ_aws))
+                    except:
+                        raise InvalidSourceOwner(princ_aws)
+
+            for account_number in account_numbers:
+                self._check_account(account_number, sqsitem, 'policy')
+                
+    def _check_account(self, account_number, sqsitem, source):
+        account = Account.query.filter(Account.number == account_number).first()
+        account_name = None
```

Comment:
`SRC [None] DST [test]. Location: policy`

I'd prefer if that would show the account number of the unknown account instead of None.  Could also condense a few lines into one:

account_name = account.name if account else account_number